src/main/java/sirius/biz/tycho/academy/OnboardingController.java

```diff
@@ -0,0 +1,155 @@
+        }
+
+        AcademyTrackInfo trackInfo =
+                onboardingEngine != null ? onboardingEngine.fetchTrack(target, track).orElse(null) : null;
+        assertNotNull(trackInfo);
+
+        List<? extends OnboardingVideo> videos =
+                onboardingEngine != null ? onboardingEngine.fetchVideos(target, track) : Collections.emptyList();
+
+        webContext.respondWith()
+                  .template("/templates/biz/tycho/academy/track.html.pasta",
+                            videos,
+                            trackInfo,
+                            target,
+                            computeURISignature(target));
+    }
+
+    /**
+     * Displays the requested video
+     *
+     * @param webContext  the request to respond to
+     * @param target      the target entity
+     * @param accessToken the security token to authenticate the target
+     * @param videoId     the id of the video to show
+     */
+    @Routed(value = "/academy/:1/:2/video/:3", priority = 999)
+    public void video(WebContext webContext, String target, String accessToken, String videoId) {
+        if (!verifyURISignature(webContext, target, accessToken)) {
+            return;
+        }
+
+        assertNotNull(onboardingEngine);
+
+        OnboardingVideo video = onboardingEngine.fetchVideo(target, videoId);
+        assertNotNull(video);
+
+        List<? extends OnboardingVideo> otherRecommendations =
+                onboardingEngine.fetchOtherRecommendations(target, videoId, video.fetchAcademyVideoData().getTrackId());
+
+        Tuple<Extension, AcademyProvider> settings =
+                onboardingEngine.fetchAcademySettings(video.fetchAcademyVideoData().getAcademy());
+
+        webContext.respondWith()
+                  .template("/templates/biz/tycho/academy/video.html.pasta",
+                            video,
+                            otherRecommendations,
+                            target,
+                            accessToken,
+                            settings.getFirst(),
+                            settings.getSecond());
+    }
+
+    /**
+     * Provides a JSON API to update the statistics which record which video has been viewed or skipped.
+     *
+     * @param webContext  the request to respond to
+     * @param out         the JSON response
+     * @param target      the target entity
+     * @param accessToken the security token to authenticate the target
+     * @param videoId     the id of the onboarding video to update
+     */
+    @Routed(value = "/academy/:1/:2/update/:3", priority = 999)
+    @InternalService
+    public void updateVideo(WebContext webContext,
+                            JSONStructuredOutput out,
+                            String target,
+                            String accessToken,
+                            String videoId) {
+        if (!verifyURISignature(webContext, target, accessToken)) {
+            return;
+        }
+        if (onboardingEngine == null) {
```

Comment:
Why is this class a mixture of all possible ways to handle when onboardingEngine is null? Some routes handle it gracefully with default/fallback values, some assert and fail and some just do nothing. Can this be cleaned up a bit?